projects/core/modules/notifications/notification-alert/Notification-alert.ts

```diff
@@ -8,15 +9,15 @@ import {
 } from '../notification-options';
 
 export class NotificationAlert<O, I> {
-    readonly status: TuiNotification;
+    readonly status?: TuiNotification;
```

Comment:
Let's not change these to optional but instead let's inject the defaults into `TuiNotificationsService` and use it in `show` method:
```ts
const notification = new NotificationAlert(observer, content, {...this.options, ...options});
```
Also a good TODO for v3 is to make all keys of options interface mandatory and use `Partial` utility type where they all are optional. Right now we can add `Required` utility type to constructor parameter in this file.

Reply:
I wonder if it also reads better to make options `private readonly` and use them here instead of restructuring on constructor 🤔
```ts
readonly hasIcon = this.options.hasIcon;

readonly autoClose = this.options.autoClose;
//...
```